lib/rdoc/token_stream.rb

```diff
@@ -65,7 +59,11 @@ def self.to_html(token_stream)
       text = CGI.escapeHTML text
 
       if style then
-        "<span class=\"#{style}\">#{text}</span>#{"\n" if comment_with_nl}"
+        if text.end_with?("\n")
+          newline = "\n"
+          text = text.byteslice(0...-1)
+        end
+        "<span class=\"#{style}\">#{text}</span>#{newline}"
```

Comment:
It took me a while to realise that we're utilizing `newline` being `nil` for not generating newlines, which only exists because we declare `newline` conditionally.
Can we add a `newline = nil` in the beginning of the method?